src/boot/bootctl.c

```diff
@@ -309,6 +309,34 @@ static int status_variables(void) {
         return 0;
 }
 
+static int boot_entry_file_check(const char *root, const char *p) {
+        _cleanup_free_ char *path;
+
+        path = path_join(root, p);
+        if (!path)
+                return log_oom();
+
+        if (access(path, F_OK) < 0)
+                return -errno;
+
+        return 0;
+}
+
+static void boot_entry_file_list(const char *field, const char *root, const char *p) {
+        int status = boot_entry_file_check(root, p);
+        const char
+                *on =    status == 0 ? "" : ansi_highlight_red(),
+                *off =   status == 0 ? "" : ansi_normal(),
+                *open =  status == 0 ? "" : " (",
+                *msg =   status == 0 ? "" : strerror(-status),
```

Comment:
we probably should not add new uses of strerror() to our codebase, since it's not thread safe. in this case it's even easy to avoid, just use `%m` in the format string, and set `errno = -status;` first...

Reply:
Well, the man page says it can only be overwritten by another strerror call, and anyway, we know we are single threaded. But OK, I'll change this to %m.